client/components/domains/domain-search-results/index.jsx

```diff
@@ -135,48 +137,97 @@ class DomainSearchResults extends Component {
 				offer = null;
 			}
 
-			let domainUnavailableMessage = [ TLD_NOT_SUPPORTED, UNKNOWN ].includes( lastDomainStatus )
-				? translate(
-						'{{strong}}.%(tld)s{{/strong}} domains are not available for registration on WordPress.com.',
+			let domainUnavailableMessage;
+
+			if ( ! isSubdomain( domain ) ) {
+				domainUnavailableMessage = [ TLD_NOT_SUPPORTED, UNKNOWN ].includes( lastDomainStatus )
+					? translate(
+							'{{strong}}.%(tld)s{{/strong}} domains are not available for registration on WordPress.com.',
+							{
+								args: { tld: lastDomainTld },
+								components: {
+									strong: <strong />,
+								},
+							}
+					  )
+					: translate(
+							'{{strong}}%(domain)s{{/strong}} is already registered. {{a}}Do you own it?{{/a}}',
+							{
+								args: { domain },
+								components: {
+									strong: <strong />,
+									a: (
+										// eslint-disable-next-line jsx-a11y/anchor-is-valid
+										<a
+											href="#"
+											onClick={ this.props.onClickUseYourDomain }
+											data-tracks-button-click-source={ this.props.tracksButtonClickSource }
+										/>
+									),
+								},
+							}
+					  );
+
+				if ( isDomainOnly && ! [ TLD_NOT_SUPPORTED, UNKNOWN ].includes( lastDomainStatus ) ) {
+					domainUnavailableMessage = translate(
+						'{{strong}}%(domain)s{{/strong}} is already registered. Do you own this domain? {{a}}Transfer it to WordPress.com{{/a}} now, or try another search.',
 						{
-							args: { tld: lastDomainTld },
+							args: { domain },
 							components: {
 								strong: <strong />,
+								a: (
+									// eslint-disable-next-line jsx-a11y/anchor-is-valid
+									<a href={ `/setup/domain-transfer?new=${ domain ?? '' }` } />
+								),
 							},
 						}
-				  )
-				: translate(
-						'{{strong}}%(domain)s{{/strong}} is already registered. {{a}}Do you own it?{{/a}}',
+					);
+				}
+			} else {
```

Comment:
It seems this `else` block is identical to the `if` block, except for the `! [ TLD_NOT_SUPPORTED, UNKNOWN ].includes( lastDomainStatus )` condition, is that right?

May I suggest a refactor to reduce duplicated code? I've uploaded it here 32df4-pb 🙂